### PR TITLE
feat: display message sending failure

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- The failure of sending the message was not updated on the UI

### Solution

- Correctly handle the status of the Message coming out of the use case observing the local message status

### Dependencies (Optional)

-This PR can be merged directly, but will not work once we merge the one on Kalium side : https://github.com/wireapp/kalium/pull/412

image

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-517/1/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 1]
```diff
@@ -201,25 +205,47 @@ private fun AnnotatedString.Builder.appendBody(messageBody: MessageBody) {
...
+                style = LocalTextStyle.current.copy(color = MaterialTheme.wireColorScheme.labelText)
+            )
+        }
+    } else {
+            Row {
+                Text(
+                    text = stringResource(id = messageStatus.stringResourceId),
+                    style = LocalTextStyle.current.copy(color = MaterialTheme.colorScheme.error)
+                )
+                Spacer(Modifier.width(4.dp))
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt:238]: I think we should use `dimensions().spacing4x` in this fixed dp values :D

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 2]
```diff
@@ -201,25 +205,47 @@ private fun AnnotatedString.Builder.appendBody(messageBody: MessageBody) {
...
+                .border(
+                    BorderStroke(
+                        width = 1.dp,
+                        color = MaterialTheme.wireColorScheme.divider
+                    ),
+                    shape = RoundedCornerShape(size = 4.dp)
+                )
+                .padding(
+                    horizontal = 4.dp,
+                    vertical = 2.dp
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt:224]: I think we should use `dimensions()` in this fixed dp values :D

**Review by yamilmedina — APPROVED** [review]
Great job fine-tuning this component 👏 ! Just left some comments about using `dimensions()` in some fixed dp values 🚀

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-517/3/) **failed**.

**Review by ohassine — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-517/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-517/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3140915)(3140915)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3140915)(3140915)-dev-release.apk)

**Pull request merged**